server.js

```diff
@@ -19,11 +39,42 @@ app.get("*", function(req, res) {
   res.sendFile(path.join(__dirname, "index.html"));
 });
 
-app.listen(3000, "localhost", function (err) {
-  if (err) {
-    console.log(err);
-    return;
-  }
+wss.on('connection', function connection(ws) {
+  const location = url.parse(ws.upgradeReq.url, true);
+  // you might use location.query.access_token to authenticate or share sessions
+  // or ws.upgradeReq.headers.cookie (see http://stackoverflow.com/a/16395220/151312)
+  ws.on('message', function incoming(message) {
+    const payload = JSON.parse(message);
+    // console.log('received: ', util.inspect(message, false, null));
+    state = payload;
+    wss.clients.forEach(function (client) {
+      if (client !== ws) client.send(message);
+    });
+  });
 
-  console.log("Listening at http://localhost:3000");
+  ws.on('close', function close() {
+    console.log('socket disconnected');
+  });
+
+  ws.send(
+    JSON.stringify(state)
+  );
+  /**
+  state.forEach(function (op) {
```

Comment:
Actually that is cruft that I should have removed before making the commit. My fault there. Anyways this pull request will need to be refactored in light of the pending removal of the spectacle tag.